recipes/ceres-solver/all/conanfile.py

```diff
@@ -120,6 +120,14 @@ def _configure_cmake(self):
         return self._cmake
 
     def build(self):
+        # In ceres 2.0.0, there is a problem with checking eigen version
+        # on ARM architecture.  This has been fixed in
+        # https://github.com/ceres-solver/ceres-solver/commit/323c350a66c503fe2815286f48da391b6ce01bc6
+        # but, at  the time of writing, no ceres release contains the
+        # fix. We apply a quick patch here to fix ARM build.
+        tools.replace_in_file(self._source_subfolder + '/CMakeLists.txt',
+                'EIGEN3_VERSION_STRING', "Eigen3_VERSION", strict=False)
```

Comment:
```suggestion
        tools.replace_in_file(os.path.join(self._source_subfolder, "/CMakeLists.txt"),
                "EIGEN3_VERSION_STRING", "Eigen3_VERSION", strict=False)
```

Reply:
Please move this fix in a patch specific to 2.0.0, to keep conanfile.py as clean as possible (sometimes it's not possible, but here it is).
tools.replace_in_file with strict=False should be avoided I believe, it leads to warnings in others versions. I prefer to be very explicit on which versions require a modification or not.

Reply:
Good point! Done. Right, it is better not to touch conanfile.py.